### ISSUE TITLE
UART causes system to stop working

### ISSUE THREAD

**Issue by ssieb** [esphome/issues#706]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):** pip and local build

**ESP (ESP32/ESP8266, Board/Sonoff):** ESP8266 tywe1s module

**Affected component:** uart

**Description of problem:**
If I include a uart section in the yaml, the device stops functioning.  Without it, the device connects to the wifi and I can see the logs.  If I add it, then there is no attempt to connect.  I've tried disabling the logger, I've tried different RX and TX pins, but it doesn't make any difference.  I tried to get the logger to use the uart, but that didn't seem to work either.  The uart is working because I can use it to flash the firmware.

**Problem-relevant YAML-configuration entries:**
```yaml
  name: dim1
  platform: ESP8266
  board: esp01_1m

wifi:
  ssid: "my ssid"
  password: "password"

# Enable logging
logger:
  baud_rate: 0

# Enable Home Assistant API
api:

ota:

#uart:
#  rx_pin: GPIO13
#  tx_pin: GPIO15
#  baud_rate: 9600

```

**Logs (if applicable):**

```
no logs available

```

**Additional information and things you've tried:**

My end goal is to implement a Tuya component, so I would like to be able to use the hardware uart on pins 13 and 15.  It's not clear how to make that work.  The documentation says the logger can use that, but somehow it's shared.  If the logger is going to write things on that connection, it won't work.

**Comment by ssieb** [comment]
From the OTA logs when it does work, there is this section:
```
[17:52:45][C][logger:175]: Logger:
[17:52:45][C][logger:176]:   Level: DEBUG
[17:52:45][C][logger:177]:   Log Baud Rate: 115200
[17:52:45][C][logger:178]:   Hardware UART: UART0
```
However, if I try to access the logs from the serial port, there is nothing.

**Comment by ssieb** [comment]
Ok, apparently I didn't get the right combination before.  I set the logger baud rate to 0, set the uart to GPIOs 13 and 15 and now it works.
This device is very strange though.  I can't get serial out through the default pins even though programming works.  I've tried various firmwares and directly programming my own arduino sketch.  But now that I can get it to boot with the uart enabled, I won't worry about that since those pins will never be needed.
